keps/sig-apps/3328-add-queueName-to-jobSpec/README.md

```diff
@@ -0,0 +1,866 @@
+e.g. in [Kubeflow](https://github.com/kubeflow/kubeflow),
+we can specify the queue name of the distributed training job in [`SchedulingPolicy`](https://github.com/kubeflow/common/blob/21f5ba8833a2e21df17601497a08396c9bae9ab2/pkg/apis/common/v1/types.go#L202-L209)
+We can also found the same implementation in [kubedl](https://github.com/kubedl-io/kubedl),
+another project helps to run deep learning workloads.
+
+[Kueue](https://github.com/kubernetes-sigs/kueue), a Kubernetes native Job queueing management is also
+calling out this feature. Currently a Kubernetes Job references to the kueue via the annotation,
+which is not native.
+
+So it's a good choice to support queueName in Job API, which helps to unify the usage for batch users.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Add a new field `queueName` to the spec of Job
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- How to integrate with non-Kubernetes native Job controllers, e.g. TFJob Operator
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+Add a new field `queueName` to the spec of Job to unify the usage of batch users.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+I'm going to develop a queue controller which helps me to manage the Jobs in resource quota,
+multi-tenant, and execution sequence. The feature supports of current Job API is excellent,
+but I found it's not graceful to bind the Jobs to the queue, the easiest way is via the annotation.
+I hope we have a native way tying them together.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+None
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+None.
```

Comment:
I would put as a risk that users might think that kubernetes does something with the queue name. And that is solved through documentation.